packages/date/src/index.js

```diff
@@ -440,4 +477,51 @@ export function getDate( dateString ) {
 	return momentLib.tz( dateString, WP_ZONE ).toDate();
 }
 
+/**
+ * Creates a moment instance using the given timezone or, if none is provided, using global settings.
+ *
+ * @param {Date|string|Moment|null} dateValue Date object or string, parsable
+ *                                            by moment.js.
+ * @param {string|number|null}      timezone  Timezone to output result in or a
+ *                                            UTC offset. Defaults to timezone from
+ *                                            site.
+ *
+ * @see https://en.wikipedia.org/wiki/List_of_tz_database_time_zones
+ * @see https://en.wikipedia.org/wiki/ISO_8601#Time_offsets_from_UTC
+ *
+ * @return {Moment} a moment instance.
+ */
+function buildMoment( dateValue, timezone = '' ) {
+	const dateMoment = momentLib( dateValue );
+
+	if ( timezone && ! isUTCOffset( timezone ) ) {
+		return dateMoment.tz( timezone );
+	}
+
+	if ( timezone && isUTCOffset( timezone ) ) {
+		return dateMoment.utcOffset( timezone );
```

Comment:
Sorry, one more thing I notice here: `utcOffset` doesn't seem to handle the same inputs we allow through `isUTCOffset`. Notably, we seem to match timezone offsets in formats like `-04:00` through `isUTCOffset`, but those same values passed through `utcOffset` would return `undefined`.

```
moment.tz.zone('America/Los_Angeles').utcOffset('-04:00');
```

Following usage documentation here, it doesn't seem like `utcOffset` accepts anything other than a numeric timestamp?

https://momentjs.com/timezone/docs/#/zone-object/offset/

Reply:
Ah, I think I was getting confused between `moment.utcOffset` and `moment.tz.utcOffset`, which have different signatures.

https://momentjs.com/docs/#/manipulating/utc-offset/
https://momentjs.com/timezone/docs/#/zone-object/offset/

It's confusing 😬 But I think it's correctly used here after all.